### ISSUE TITLE
Check peer dependencies for a reflect polyfill upon install

### ISSUE THREAD

**Issue by Xapphire13** [Microsoft/tsyringe#18]
This package has an implicit peer dependency on a reflect polyfill, we should check to see if one is installed in the `postinstall`. If not we could output a message like:

```
[WARN] tsyringe requires a reflect polyfill, but none is installed. Consider installing one of "reflect-metadata", "core-js", or "reflection".
```

**Comment by Xapphire13** [comment]
Not sure on how feasible this is. `npm` will show the output warnings, however `yarn` hides it unless you have `--verbose` specified.

There is also complications with detecting if the package was installed in a nested hierarchy, or a flattened one.

**Comment by megahertz** [comment]
Maybe it's better to add runtime check in index.ts, like
```ts
if (typeof Reflect === "undefined" || !Reflect.getMetadata) {
  throw `tsyringe requires a reflect polyfill. Please add import "reflect-metadata" to your entry point.`;
}
```

Even if a polyfill is installed, it doesn't guaranty that user is actually required it.

**Comment by Xapphire13** [comment]
Sounds good to me. Would you like to submit a PR?

**Comment by MeltingMosaic** [comment]
Closed by #24
